lib/cocaine/command_line/multi_pipe.rb

```diff
@@ -29,19 +29,51 @@ def close_write
       end
 
       def read
-        @stdout_output = read_stream(@stdout_in)
-        @stderr_output = read_stream(@stderr_in)
+        read_streams(@stdout_in, @stderr_in)
       end
 
       def close_read
-        @stdout_in.close
+        begin
+          @stdout_in.close
+        rescue IOError
+          # do nothing
```

Comment:
Are you sure we don't need to do anything here? Under what circumstances might this raise and what does it mean for the calling code?

Reply:
when a program exits and we detect it on the error, closing can cause an exception. IOError in close isn't fixable, so I see no reason to do anything with it.